compositor_pipeline/src/queue/audio_queue.rs

```diff
@@ -173,7 +173,16 @@ impl AudioQueueInput {
             .queue
             .iter()
             // start_pts and end_pts are already in units of this input
-            .filter(|batch| batch.start_pts <= end_pts && batch.end_pts >= start_pts)
+            .filter(|batch| {
+                let batch_in_range = batch.start_pts <= end_pts && batch.end_pts >= start_pts;
+                match self.offset {
+                    Some(_offset) => batch_in_range,
+                    // We need to check if batch.start_pts does not represent time before queue
+                    // start. PTS can only be positive, so without this check we could produce
+                    // incorrect start_pts value on output framerate.
```

Comment:
No idea what you meant here:
`we could produce incorrect start_pts value on output framerate.`

I guess you meant:
`we could produce incorrect start_pts value on output.`

Still, handling this here sounds like mixing logic for handling specific corner cases with common / "expected" flow.

Reply:
> Still, handling this here sounds like mixing logic for handling specific corner cases with common / "expected" flow.

I don't understand this point?